### ISSUE TITLE
accuracy: Create new plugin type of scoring models

### ISSUE THREAD

**Issue by pdxjohnny** [intel/dffml#732]
We'll do a review at the end of each phase. For each phase, we'll open a PR against the `accuracy_scorer` branch (rather than `master`)

### Phase 1 - Initial Implementation

- [x] Create a new plugin type, Accuracy
- [x] Each entry will have a
  - [x] CONFIG
  - [x] Score coroutine method (`async def`)
    - [x] Inputs
      - [x] First is the same as the output of the predict method (an asynciterator of Record objects)
      - [x] Second is list of Feature instances that we did prediction on
    - [x] The method will look at the record.features() for the features we predicted, and the record.predictions() and return it's assessment of the accuracy.
- [x] Change accuracy method of model to no longer be an `abstractmethod`, we'll make it so that it takes a second parameter, the accuracy_scorer (of type `AccuracyContext`). The `accuracy` method within `ModelContext` will
  - [x] check if `self.config.predict` exists using `hasattr`, if not raise `NotImplementedError`
  - [x] return a call to `accuracy_scorer.score(self.predict(sources), [self.config.predict])`
- [x] We will need to modify the high level accuracy function to take accuracy_scorer as the second argument (after `model`)
- [x] We will need to implement at least one of these from scratch within the main package (dffml) so that we can use it for testing and examples and the like

### Phase 2 - Documentation

- [x] Add and example with tests for the MSE accuracy scoring plugin
- [x] Modify `dffml/model/slr.py` to remove the accuracy method
- [x] Modify `scripts/docs/care` to add a `dffml.accuracy dffml` line

### Phase 3 - Update Model Plugins

  - [x] Remove `accuracy` method from the following models
    - [x] dffml-model-scratch
    - [x] dffml-model-scikit
    - [x] dffml-model-tensorflow
    - [x] dffml-model-vowpalWabbit
    - [x] dffml-model-autosklearn
    - [x] dffml-model-daal4py
  - [x] For NLP models modify accuracy method signature to accept a scorer but log.warning that it won't be used
    - [x] dffml-model-tensorflow-hub
    - [x] dffml-model-transformers

### Phase 3.33 - Update Model Plugin - Merge New Models from master branch

- [x] Create PR that merges new models from master into accuracy scorer branch

### Phase 3.66 - Update Model Plugin - Again

  - [x] Remove `accuracy` method from the following models
    - [x] dffml-model-pytorch
  - [x] For NLP models modify accuracy method signature to accept a scorer but log.warning that it won't be used
    - [x] dffml-model-spacy

### Phase 4 - Update Tests and Examples

- [x] Update `dffml/cli/ml.py` first so that CLI tests and examples will be able to specify which accuracy scorer to use
- [x] Update tests and examples accordingly, use the MSE accuracy scorer for everything except ones where we took out the accuracy method in phase 3

### Phase 5 - Update AccuracyScorer.score signature

- [x] Change all of the `score` methods within all the `AccuracyScorer`s so that their signature is as follows
  - `def score(self, mctx: ModelContext, sources: SourcesContext):`
- [x] Remove the `accuracy` method from all of the `ModelContext`s
  - There should one to remove from the base `ModelContext`
  - There should be another to remove in `model/tensorflow_hub/dffml_model_tensorflow_hub/text_classifier.py`
- [x] Modify all of tests that don't use the `high_level:accuracy` function to now use that function
  - Anywhere we were doing the double context entry from when we didn't have the high level function (for example [here](https://github.com/intel/dffml/blob/1969195b1165c1bf9102deb4127b6f5517455d3d/model/tensorflow_hub/tests/test_model.py#L60-L63)) we want to now use the high_level accuracy function.
- [x] Modify the `high_level.py:accuracy` function and the `no_async.py:accuracy` function so that they call the score method appropriately.

### Phase 6 - Update HTTP Service

- [x] The HTTP service will need updating to understand accuracy scorers
  - [x] Add `configure_` routes for accuracy scorers
  - [x] Tests for `configure_` routes for accuracy scorers
- [x] Remove model accuracy route. Replace with /scorer/ route
- [x] Add CLI options (update docs/cli.rst)
  - [x] Make sure there are tests with this (tests/test_cli.py)
- [x] Update HTTP API docs
  - https://intel.github.io/dffml/master/plugins/service/http/api.html#accuracy

### Phase 7 - Rebase master

- [x] Rebase master branch. We want to avoid a merge commit since that would mess up the history.
  - First we'll merge Phase 6 into accuracy_staging
  - `git fetch origin`
  - `git checkout -b phase7 origin/accuracy_staging`
  - [x] Identify cherry picked patches if necessary
    - **We're not sure if we should to do this**
    - We may want to look through the log to see which commit messages are in both branches.
    - Drop the cherry picked commits from this branch
  - `git rebase origin/master`

### Phase 8 - Update Documentation (Tutorials, Use Cases, CLI, etc.)

- [x] Update Documentation that isn't being tested within the CI, check docs in the following locations
  - tutorials
   - Use Cases, CLI, etc.)
